kairon/shared/data/collection_processor.py

```
@@ -1,9 +1,231 @@
+            collection_name = item.pop('collection_name', None)
+            is_secure = item.pop('is_secure')
+            is_non_editable = item.pop('is_non_editable')
+            data = item.pop('data')
+            data = DataProcessor.prepare_decrypted_data(data, is_secure)
+            final_data["_id"] = item["_id"].__str__()
+            final_data['collection_name'] = collection_name
+            final_data['is_secure'] = is_secure
+            final_data['is_non_editable'] = is_non_editable
+            final_data['data'] = data
+        except DoesNotExist:
+            raise AppException("Collection data does not exists!")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)